cmd/gapic-showcase/compliance_suite_negative_test.go

```diff
@@ -0,0 +1,116 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"testing"
+
+	"github.com/googleapis/gapic-showcase/server/genproto"
+	pb "github.com/googleapis/gapic-showcase/server/genproto"
+	"google.golang.org/protobuf/encoding/protojson"
+)
+
+// TestComplianceSuiteErrors checks for non-spec-compliant HTTP requests. Not all of these
+// conditions necessarily generate a server error in a real service, but the behavior is often
+// ill-defined. We want Showcase to require the generators be strict in the transcoding format they
+// use.
+func TestComplianceSuiteErrors(t *testing.T) {
```

Comment:
Nit: can we rename the file to `compliance_suite_errors_test.go` to match up with the main test function present? I think that is better name anyways, but I'm not too picky.